README.md

```diff
@@ -1,145 +1,231 @@
 
-You can now create a react component with your pug inside it.
+    ```
+    {
+      "plugins": [
+        "transform-react-pug",
+        "transform-react-jsx"
+      ]
+    }
+    ```
 
-```js
-import React from 'react';
+3.  Now all your templates written with pug are understood by react and browsers.
 
-class MyComponent extends React.Component {
+### create-react-app
 
-  render() {
-    return pug`
-      div
-        h1 My Component
-        p This is my component using pug.
-    `;
-  }
-}
-```
+Integrating with [create-react-app][link to cra] is tricky because it does not allow you to modify babel configuration. There are two documented possibilities:
+
+* [eject][link to cra eject]
 
-### Example 2 - Re-using a Pug Component
+  That is easy, you will get `.babelrc` file in your root directory, just add `transform-react-pug` before `transform-react-jsx` there.
 
-You can use a pug component in another component.
+* [react-app-rewired][link to rewired cra]
 
-```js
-import React from 'react';
-import MyComponent from './my-component'
+  Go through official instruction to rewire your application. Then modify your `config-overrides.js`:
 
-class MyNewComponent extends React.Component {
+  ```diff
+  + const {injectBabelPlugin} = require('react-app-rewired');
+    module.exports = function override(config, env) {
+  -   //do stuff with the webpack config...
+  +   config = injectBabelPlugin('transform-react-pug', config);
+      return config;
+    }
+  ```
 
-  render() {
+[link to cra]: https://github.com/facebook/create-react-app/
+[link to cra eject]: https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#npm-run-eject
+[link to rewired cra]: https://github.com/timarney/react-app-rewired#1-injectbabelplugin
 
-    const prop1 = 'This is something to pass to another component';
+### React Native
 
-    return pug`
-      div
-        h1 MyNewComponent
-        p This component imports my other component.
-        p It could import several of these within the pug.
-        MyComponent
+Just add this plugin to the list in `.babelrc` file.
 
-        p If I had created a component with props I could pass them from this component.
-        AComponentExpectingProps(
-          prop1 = prop1
-        )
-    `
+```diff
+  {
+-   "presets": ["react-native"]
++   "presets": ["react-native"],
++   "plugins": ["transform-react-pug"]
   }
-}
 ```
 
-### Example 3 - Creating a Pug Constant
+_We don't need `transform-react-jsx` here because it's coming with `react-native` preset._
 
-You can create a pug constant that you can simply re-use in your code.
+## How it works
 
-```js
-import React from 'react';
+_Coming soon..._
 
-class MyComponent extends React.Component {
+## Limitations
 
-  _onDoOneThing = () => {
-    console.dir('Do one thing');
-  };
+* No possibility to use static methods. We have to use `Fragment` instead of `React.Fragment`
```

Comment:
I wonder if this can be written more clearly?

Reply:
@ForbesLindesay my initial intention was to describe that we can't use dots in component names. React.Fragment was a great example, but now I see that it's probably not.

What do you think about these options:

>We can't use dots in component names because pugjs understands them like className.

>We can't use dots in component names because pugjs treats everything after dot like className. For example, `React.Fragment` will be equal to `<React className="Fragment" />`

If neither of those works, I think I need your help to demonstrate this limitation.

Reply:
> We can't use dots in component names because pugjs treats everything after dot as a className. For example, `React.Fragment` becomes `<React className="Fragment" />`, not `<React.Fragment />`